### ISSUE TITLE
default/missing value are passed in deserialized form with marshmallow 3

### ISSUE THREAD

**Issue by lafrech** [marshmallow-code/apispec#342]
This will break automatic default documentation:

```py
    def field2property(self, field, use_refs=True, dump=True, load=True, name=None):
        [...]
        if 'doc_default' in field.metadata:
            ret['default'] = field.metadata['doc_default']
        else:
            default = field.missing
            if default is not marshmallow.missing and not callable(default):
                ret['default'] = default
```

Since we have the field object, maybe we could use it to dump the value for the doc?

Could that fail in some corner case?

**Comment by lafrech** [comment]
Dumping the `missing` value could be wrong if the field does not round-trip.

In the general case, it's fine, but could there be example of fields where dump(load(value)) != value?

I can't think of a simple example, but it could happen in a custom field, or in a `Nested` field where the schema has complicated logic with pre/post load decorators.

It could also be wrong if a field behaves differently depending on the context. Again, `Nested` is the typical example for this as a schema can embed any sort of processing.

Also, a `Nested` field with a `missing` value of `{}` and a schema with `default` values in fields would generate a wrong documentation if we call nested_field.dump(`{}`).

As soon as we call `dump`, we're more or less in the callable case, and the safest thing to do would be to do nothing.

OTOH, in most cases, dumping `missing` is perfectly fine, so removing automatic default documentation would be a regression. Expecting the user to specify `doc_default` negates the benefits of the change in MA3 allowing to pass values in object form.

For now, we probably need to at least inhibit the feature in MA3 to avoid values in object form (potentially not json-serializable) being injected in the spec dict without being serialized.

**Comment by lafrech** [comment]
Fixed by #490.
